Add ability to pass annotations to the metriton reporter.

thallgren opened the pull request:
## Description

Adds the `ReportAnnotations` extension point so that other modules can add dynamic data to the metriton reports.

## Checklist

 - [ ] I made sure to update `./CHANGELOG.md`.
 - [ ] I made sure to add any docs changes required for my change (including release notes).
 - [x] My change is adequately tested.
 - [ ] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.
 - [ ] I updated `TELEMETRY.md` if I added, changed, or removed a metric name.
 - [ ] Once my PR is ready to have integration tests ran, I posted the PR in #telepresence-dev in the datawire-oss slack so that the "ok to test" label can be applied.

Commit:
Add ability to pass annotations to the metriton reporter.

Adds the `ReportAnnotations` extension point so that other modules can
add dynamic data to the metriton reports.

Signed-off-by: Thomas Hallgren <thomas@datawire.io>

raphaelreyna commented on pkg/client/scout/reporter.go:
```diff
@@ -181,15 +184,19 @@ func NewReporterForInstallType(ctx context.Context, mode string, installType Ins
 }
 
+// ReportAnnotators are the default annotator functions that the NewReporter function will pass to NewReporterForInstallType.
+var ReportAnnotators []ReportAnnotator //nolint:gochecknoglobals // extension point
```
might be nice to name this `DefaultReportAnnotators`

raphaelreyna commented on pkg/client/scout/reporter.go:
```diff
@@ -200,7 +207,7 @@ func (r *Reporter) initialize(ctx context.Context, mode, goos, goarch string) {
 	baseMeta := getOsMetadata(ctx)
 	baseMeta["mode"] = mode
-	baseMeta["trace_id"] = uuid.New()
+	baseMeta["trace_id"] = uuid.New().String() //  It's sent as JSON so might as well convert it to a string once here.
```
`uuid.NewString()` is also available

raphaelreyna reviewed (approved):
lgtm; just two minor nitpicks.

Commit:
Address review feedback

- ReportAnnotators -> DefaultReportAnntoators
- UUID.New().String() -> UUID.NewString()

Signed-off-by: Thomas Hallgren <thomas@datawire.io>
